From 597110a2bcd141ac5c3a532874286d99b2e23f38 Mon Sep 17 00:00:00 2001
From: "David C. Manuelda" <StormByte@gmail.com>
Date: Thu, 3 Sep 2026 06:29:46 +0200
Subject: [PATCH] cmake: pkg-config stem matches the installed jpeg archive

MSVC static installs jpeg-static.lib; Unix installs jpeg.
FFmpeg Meson was looking for jpeg.lib on Windows.
---
 cmakescripts/BuildPackages.cmake | 6 ++++++
 release/libjpeg.pc.in            | 2 +-
 2 files changed, 7 insertions(+), 1 deletion(-)

diff --git a/cmakescripts/BuildPackages.cmake b/cmakescripts/BuildPackages.cmake
index 4f55f146..2c4b9244 100644
--- a/cmakescripts/BuildPackages.cmake
+++ b/cmakescripts/BuildPackages.cmake
@@ -174,6 +174,12 @@ configure_file(release/maketarball.in pkgscripts/maketarball)
 add_custom_target(tarball pkgscripts/maketarball
   SOURCES pkgscripts/maketarball)
 
+# Same stem BM installs. Meson looks at this line, not at CMake Config.
+if(WIN32 AND NOT ENABLE_SHARED)
+    set(JPEG_PC_LIB "jpeg-static")
+else()
+    set(JPEG_PC_LIB "jpeg")
+endif()
 configure_file(release/libjpeg.pc.in pkgscripts/libjpeg.pc @ONLY)
 
 if(WITH_TURBOJPEG)
diff --git a/release/libjpeg.pc.in b/release/libjpeg.pc.in
index 74fb7fc6..d0e1d3f3 100644
--- a/release/libjpeg.pc.in
+++ b/release/libjpeg.pc.in
@@ -6,5 +6,5 @@ includedir=@CMAKE_INSTALL_FULL_INCLUDEDIR@
 Name: libjpeg
 Description: A SIMD-accelerated JPEG codec that provides the libjpeg API
 Version: @VERSION@
-Libs: -L${libdir} -ljpeg
+Libs: -L${libdir} -l@JPEG_PC_LIB@
 Cflags: -I${includedir}
-- 
2.55.0

